### PR TITLE
Support custom ModelManagerException messages

### LINKED ISSUE #8137: Customise error message for ModelManagerException
## Feature Request

Following from #7759, it is not possible to use the feature when the exception is of type ModelManagerException, even though these implement ModelManagerThrowable, as is the case for Doctrine UniqueConstraintViolationExceptions.

This is because, in CRUDController.php ModelManagerExceptions are caught first, before the catch for ModelManagerThrowable.

The relevant catches have been marked for removal for the last couple of years `// NEXT_MAJOR: Remove this catch.` but persist in the current 4.x and 5.x branches.

Currently, I'm having to hack around this issue by overriding addFlash, disabling the call to the CRUDController method for cases where I want to customise the message and storing the custom message in the admin (if there is one) so that I can retrieve it in the controller, before re-enabling the call to the CRUDController afterwards.

I see three possible clean resolutions to this:
1. remove the catch blocks that have been earmarked since 2021 (this may have B.C. issues for some projects);
2. swap the ordering of the ModelManagerThrowable and ModelManagerException catch blocks so that ModelManagerThrowable is handled first;
3. apply a similar interface to handleModelManagerException so that it can also return a customised string.

I prefer option 3... is there any reason why this was not done at the same time as #7761 ?

### PR DESCRIPTION
**Pull request by wadjei**
## Support custom ModelManagerException messages

Add the option of customising the message passed to the FlashBag for caught `ModelManagerException` by returning a 
string from your overridden handleModelManagerException() methods.

This is similar to what was done previously for handleModelManagerThrowable() but that code path seems to
be unreachable.

The method signature for `handleModelManagerException` has had its return type removed.

New tests have been added to validate the changes work as intended by way of two new controllers in the `Sonata\AdminBundle\Tests\App\Controller` namespace: `CustomModelManagerThrowableMessageController` and `CustomModelManagerExceptionMessageController`.

I am targeting this branch, because the change addresses an apparent inability to use handleModelManagerThrowable customised message facility from #7761.

Closes #8137

## Changelog

```markdown
### Changed
CRUDController::handleModelManagerException can now return a custom error message string for display in the flashbag instead of Sonata ones. Return type is removed from the signature

```

### CODE REVIEW

**Comment by VincentLanglet** [comment]
Thanks ; can you take a look at the failing build ?

File: phpunit.xml.dist [hunk 1]
```diff
@@ -36,5 +36,6 @@ It's auto-generated by sonata-project/dev-kit package.
         <env name="KERNEL_CLASS" value="\Sonata\AdminBundle\Tests\App\AppKernel" />
         <server name="APP_ENV" value="test" force="true" />
         <server name="APP_DEBUG" value="false" />
+        <ini name="memory_limit" value="256M" />
```
Comment (VincentLanglet) [inline phpunit.xml.dist]: This file is autogenerated and should not be touch.

File: src/Controller/CRUDController.php [hunk 2]
```diff
@@ -1105,6 +1105,7 @@ protected function getBaseTemplate(): string
 
     /**
      * @throws \Exception
+     * @return string|null|void A custom error message to display in the flag bag instead of the generic one
```
Comment (VincentLanglet) [inline src/Controller/CRUDController.php]: Please change to `@return string|null` and return null at the end of the method.

**Review by VincentLanglet — COMMENTED** [review]
You also need to run `make cs-fix-php`

File: phpunit.xml.dist (hunk 1 above)
Comment (wadjei) [reply phpunit.xml.dist]: ah yes, committed by accident! The tests don'r run locally for me without this. I'll see if I can put in the environment instead

File: src/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -174,6 +174,7 @@ final class Configuration implements ConfigurationInterface
 {
     /**
      * @psalm-suppress UndefinedInterfaceMethod
+     * @psalm-suppress PossiblyNullReference - end() can return null, apparently
```
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php]:
```suggestion
```

**Review by VincentLanglet — APPROVED** [review]

**Pull request merged**